src/Storages/MutationCommands.cpp

```diff
@@ -207,6 +207,8 @@ std::shared_ptr<ASTExpressionList> MutationCommands::ast(bool with_pure_metadata
     auto res = std::make_shared<ASTExpressionList>();
     for (const MutationCommand & command : *this)
     {
+        if (!command.ast)
+            continue;
```

Comment:
When is it possible? Does it mean that some commands may be lost when serializing?

Reply:
After digging though the reflog I finally remember the reason, I was using `waitForMutation(maybe_mutation_commands)` in `StorageMergeTree::alter()` - https://github.com/azat/ClickHouse/commit/9365cc7f1355d33fc0cdf7b3e53d8dab70d116c7#diff-205c1b8d661c01d47b9831ecd03be2e25db3a49192f186a00c8896e13eac77efL381

And I cannot reproduce any crash with the current version of the patch (even with some extra debugging logging to activate `MutationsCommands::toString`), will remove this patch and rebase.